src/alexandria3k/data_source.py

```diff
@@ -749,36 +774,41 @@ def query(self, query, partition=False):
         #   Run query on in-memory database
         #   drop tables
         self.set_query_columns(query)
-        partition = apsw.Connection(
-            ":memory:", apsw.SQLITE_OPEN_READWRITE | apsw.SQLITE_OPEN_URI
-        )
-        partition.create_module("filesource", self.data_source)
-        partition.execute(
-            log_sql(f"ATTACH DATABASE '{self.vdb_uri}' AS virtual")
-        )
-
-        # Also attach databases to the partition
-        for attach_command in self.attach_commands:
-            partition.execute(log_sql(attach_command))
 
         for i in self.data_source.get_container_iterator():
             debug.log(
                 "progress",
                 f"Container {i} {self.data_source.get_container_name(i)}",
             )
+
+            # Create temporary tables in the SAME connection
             for table_name, table_columns in self.query_columns.items():
                 columns = ", ".join(table_columns)
-                partition.execute(
+                self.vdb.execute(
                     log_sql(
-                        f"""CREATE TABLE {table_name}
-                  AS SELECT {columns} FROM virtual.{table_name}
-                    WHERE virtual.{table_name}.container_id={i}"""
+                        f"""CREATE TEMP TABLE temp_{table_name} AS
+                    SELECT {columns} FROM {table_name}
+                        WHERE {table_name}.container_id={i}"""
                     )
                 )
-            self.cursor = partition.cursor()
-            yield from try_sql_execute(self.cursor, query)
+
+            # Modify the original query to use temp tables
+            temp_query = query
+            for table_name in self.query_columns:
+                temp_query = re.sub(
```

Comment:
This is brittle and can't be easily done reliably. Consider e.g. the case where the table name appears in a string.